CMakeLists.txt

```diff
@@ -1646,6 +1631,46 @@ if(WOLFSSL_CRYPTOCB)
     list(APPEND WOLFSSL_DEFINITIONS "-DWOLF_CRYPTO_CB")
 endif()
 
+if(WOLFSSL_OCSPSTAPLING)
+    set(WOLFSSL_CERTIFICATE_STATUS_REQUEST "yes")
+    list(APPEND WOLFSSL_DEFINITIONS "-DHAVE_CERTIFICATE_STATUS_REQUEST" "-DHAVE_TLS_EXTENSIONS")
+    override_cache(WOLFSSL_OCSP "yes")
+endif()
+
+if(WOLFSSL_OCSPSTAPLING_V2)
+    set(WOLFSSL_CERTIFICATE_STATUS_REQUEST_V2 "yes")
+    list(APPEND WOLFSSL_DEFINITIONS "-DHAVE_CERTIFICATE_STATUS_REQUEST_V2" "-DHAVE_TLS_EXTENSIONS")
+    override_cache(WOLFSSL_OCSP "yes")
+endif()
+
+# must be below OCSP stapling options to allow override
+if (WOLFSSL_OCSP)
+   list(APPEND WOLFSSL_DEFINITIONS "-DHAVE_OCSP")
+endif()
+
+if (WOLFSSL_CRL STREQUAL "yes")
```

Comment:
I don't think this logic is quite right. In the autotools build, we add both `HAVE_CRL` and `HAVE_CRL_IO` if given `--enable-crl=io`. This will only add `HAVE_CRL_IO`.

Reply:
Good catch. Not sure if the tests cover this case.